### PR TITLE
Bugfixes and improvements

### LINKED ISSUE #4: Use octonode methods instead of get(URL)
URLs as literals are fragile. And I'm assuming that it'll be more efficient to use `octonode`'s data structures instead of querying the GH API with a new request every time.

So, instead of:

``` javascript
client.get('/repos/w3c/' + projectName + '/commits', {}, function (err, status, body, headers) {
    ⋮
});
client.get('/repos/w3c/' + projectName + '/issues', {}, function (err, status, body, headers) {
    ⋮
});
client.get('/repos/w3c/' + projectName + '/contributors', {}, function (err, status, body, headers) {
    ⋮
});
```

do:

``` javascript
const repo = client.repo(`w3c/${projectName}`);
repo.commits((err, data, headers) => {
    ⋮
});
repo.issues((err, data, headers) => {
    ⋮
});
repo.contributors((err, data, headers) => {
    ⋮
});
```

### PR DESCRIPTION
**Pull request by tripu**
* Update npm dependencies
* Use `tripu/octonode` instead of npm's ([has updated rate limiting](https://github.com/tripu/octonode#rate-limiting))
* Use octonode methods (fixes #4)
* Code a bit more simple and succinct
* Handle errors when there are no commits/contributors/PRs/etc
* Retrieve API rate limit too; display that on console, together with time remaining until next reset
* Documentation: update and make more detailed

### CODE REVIEW

**Comment by tripu** [comment]
@gosko, care to review&hellip;?

File: package.json
Comment (tripu) [inline package.json]: Well spotted, @deniak. I think I just found a bug in that PR of mine though; I'm working on a fix for that now.

Comment (tripu) [inline package.json]: Ah. No bug. It's just that there's no updated version of `octonode` on npm. I [asked the maintainer to publish](https://github.com/pksunkara/octonode/pull/248#issuecomment-263916043). In the meantime, to avoid blocking this PR, we can refer to this specific commit.

**Comment by tripu** [comment]
Using the updated version of `octonode` now. Ready.

**Review by deniak — APPROVED** [review]

**Pull request merged**
